app/src/main/kotlin/compiler/lowlevel/Instruction.kt

```diff
@@ -152,13 +227,25 @@ sealed class Instruction : Asmable {
             abstract val reg: Register
             override val regsDefined: Collection<Register> get() = setOf(reg)
             override val regsUsed: Collection<Register> get() = setOf(reg)
+
+            override fun toAsm(registers: Map<Register, Register>) = when (this) {
+                is SetEqR -> "sete"
+                is SetNeqR -> "setne"
+                is SetLtR -> "setl"
+                is SetLtEqR -> "setle"
+                is SetGtR -> "setg"
+                is SetGtEqR -> "setge"
+            } + " ${registers[reg]}"
```

Comment:
These instructions require 8-bit register (e.g. "sete al", "sete rax" would fail), so some translation will be necessary.

Reply:
True, I added a mapping.